src/app/home-page/widget/entry-box/entry-box.component.html

```diff
@@ -43,22 +43,25 @@
     <mat-card-content class="p-3 h-100">
       <div *ngIf="totalEntries !== 0 || isLoading; else noEntries">
         <mat-form-field appearance="outline" class="search-form-field w-100 pb-0">
-          <input
-            matInput
-            data-cy="search-field"
-            [(ngModel)]="filterText"
-            (keyup)="onTextChange($event)"
-            placeholder="Search your {{ entryTypeLowerCase }}s..."
-          />
+          <mat-label>Search your {{ entryTypeLowerCase | titlecase }}s...</mat-label>
+          <input matInput data-cy="search-field" [(ngModel)]="filterText" (keyup)="onTextChange($event)" [matAutocomplete]="auto" />
+          <button *ngIf="filterText" mat-button matSuffix mat-icon-button (click)="clearSearch()" type="button">
+            <mat-icon>cancel</mat-icon>
+          </button>
         </mat-form-field>
+        <mat-autocomplete #auto="matAutocomplete" (optionSelected)="navigateToEntry(value)">
```

Comment:
😯 Thank you!!!

Two problems: `navigateToEntry` was always receiving `undefined` because I wasn't referencing the `mat-option` value correctly. Updated to:`(optionSelected)="navigateToEntry($event.option.value)"`. Secondly, need to navigate using the full `path` not `prettyPath`: `[value]="result.path"` and `this.router.navigate([this.allEntriesLink + path]);`